src/mrpro/data/_KData.py

```diff
@@ -141,12 +142,17 @@ def reshape_acq_data(data):
                     setattr(current, subfield.name, reshape_acq_data(subcurrent))
 
         # Calculate trajectory and check if it matches the kdata shape
-        if isinstance(ktrajectory, KTrajectoryCalculator):
+        if isinstance(ktrajectory, KTrajectoryIsmrmrd):
+            ktraj_raw = ktrajectory(acquisitions)
+            ktraj = ktraj_raw.reshape(sort_idx, num_k2, num_k1)
+        elif isinstance(ktrajectory, KTrajectoryCalculator):
             ktraj = ktrajectory(kheader)
         elif isinstance(ktrajectory, KTrajectory):
             ktraj = ktrajectory
         else:
-            raise TypeError(f'ktrajectory must be a KTrajectory or KTrajectoryCalculator, not {type(ktrajectory)}')
+            raise TypeError(
+                f'ktrajectory must be None, a KTrajectory or KTrajectoryCalculator, not {type(ktrajectory)}'
```

Comment:
can it be None?
ISMMRMD would be a sensible default, but currently it does not seem like it is

Reply:
No it can't be none, should be KTrajectoryIsmrmrd, KTrajectory or KTrajectoryCalculator

I would not set a default for ktrajectory because a default to me should always work, but KTrajectoryIsmrmrd would fail if the trajectory is not in the file and would yield rubbish if the trajectory is just 0s in the file.

Reply:
Ok, then just the error message is wrong